code/game/objects/items/weapons/blades.dm

```diff
@@ -82,6 +82,10 @@
 
 	var/trans
 	var/obj/item/reagent_containers/container = I
+	if(!container || !container.reagents)
+		to_chat(user, "<span class='rose'>[I] isn't compatible with [src].</span>")
+		return TRUE
```

Comment:
```suggestion
	if(!container.reagents)
		to_chat(user, "<span class='rose'>[I] isn't compatible with [src].</span>")
		return TRUE
	var/obj/item/reagent_containers/container = I
```

Reply:
Would still runtime as the typecast would fail on objects that have reagents but aren't reagent containers of which there are a few (e.g. the engineer welding backpack), leading to trying to access null.reagents.total_volume in l.89

Reply:
Typecasts can never fail in dm (a weak point actually as var/list/totallyalist = "a string" will always runtime)
It also means this typecast happens incorrectly often as any item that is a non reagent could be used to click on
Thus a better fix would be a 
if(!isreagentontainer(I))
   return check
or even more optimally finding out why I is null, as it should nver be null